lib/std/mem.zig

```diff
@@ -1706,12 +1706,24 @@ test count {
 
 /// Returns the number of needles inside the haystack
 pub fn countScalar(comptime T: type, haystack: []const T, needle: T) usize {
+    const n = haystack.len;
+    if (n < 1) return 0;
     var i: usize = 0;
     var found: usize = 0;
 
-    while (findScalarPos(T, haystack, i, needle)) |idx| {
-        i = idx + 1;
-        found += 1;
+    if (std.simd.suggestVectorLength(u8)) |block_size| {
+        const Block = @Vector(block_size, T);
+
+        const letter_mask: Block = @splat(needle);
+        while (i + block_size <= n) : (i += block_size) {
```

Comment:
Also, small suggestion(feel free to ignore)
Maybe better to write as:
```suggestion
        while (n - i >= block_size) : (i += block_size) {
```
So that it clearer to see logic - while remaining items more than block size